cli/src/cluster_query.rs

```diff
@@ -1170,6 +1219,37 @@ pub fn process_show_validators(
     Ok("".to_string())
 }
 
+pub fn process_transaction_history(
+    rpc_client: &RpcClient,
+    address: &Pubkey,
+    start_slot: Option<Slot>,
+    end_slot: Option<Slot>,
+) -> ProcessResult {
+    let (start_slot, end_slot) = {
+        if let Some(start_slot) = start_slot {
+            if let Some(end_slot) = end_slot {
+                (start_slot, end_slot)
+            } else {
+                (start_slot, start_slot + 1000)
```

Comment:
The limit we have set in rpc is 10k slots. Did you intend this to be more restrictive?

Reply:
I did but now I'm failing to find a good argument for why.   Actually I'm thinking of reworking the command to look like:
```
solana transaction-history  <ADDRESS> [end slot (default: latest slot)] [slot limit (default: 10k)]
```
where start slot would be `end slot` - `slot limit`.   I think that's maybe more intuitive